### PR TITLE
add _device attribute to PRNGKeyArray so that computation follows key placement

### PR DESCRIPTION
**Pull request by mattjj**
unrelated: remove some redundant hasattr + try / except AttributeError

### CODE REVIEW

File: jax/_src/prng.py [hunk 1]
```diff
@@ -169,6 +169,8 @@ def ndim(self):
   def dtype(self):
     return KeyTy(self.impl)
 
+  _device = property(op.attrgetter('_base_array._device'))
```
Comment (yashk2810) [inline jax/_src/prng.py:172]: Drive-by: For jax.Array you will also need to add `_committed` attributed for it to pass on the Array path.

Comment (mattjj) [reply jax/_src/prng.py:172]: Thanks!

**Review by froystig — APPROVED** [review]

**Review by yashk2810 — APPROVED** [review]
